docker-app/qfieldcloud/core/views/projects_views.py

```diff
@@ -90,21 +79,27 @@ class ProjectViewSet(viewsets.ModelViewSet):
     lookup_url_kwarg = "projectid"
     permission_classes = [permissions.IsAuthenticated, ProjectViewSetPermissions]
     pagination_class = pagination.QfcLimitOffsetPagination()
-    filter_backends = [QfcOrderingFilter]
+    filter_backends = [filters.DjangoFilterBackend, QfcOrderingFilter]
+    filterset_class = ProjectFilterSet
     ordering_fields = ["owner__username::alias=owner", "name", "created_at"]
 
     def get_queryset(self):
         projects = Project.objects.for_user(self.request.user)
 
-        # In the list endpoint, by default we filter out public projects. They can be
-        # included with the `include-public` query parameter.
         if self.action == "list":
-            include_public = False
+            # In the list endpoint, by default we filter out public projects.
+            # They can be included with the `include_public` query parameter or
+            # the deprecated `include-public` query parameter.
+            force_exclude_public = True
             include_public_param = self.request.query_params.get("include-public")
-            if include_public_param and include_public_param.lower() == "1":
-                include_public = True
+            if include_public_param and include_public_param != "":
+                force_exclude_public = False
+
+            include_public_param = self.request.query_params.get("include_public")
+            if include_public_param and include_public_param != "":
+                force_exclude_public = False
 
-            if not include_public:
+            if force_exclude_public:
```

Comment:
I would move this comments and stuff into `ProjectFilterSet.filter_include_public` with a small comment here pointing about "quirks in showing public projects".

Reply:
Nevermind, discussed over chat that this approach will not work. Thinking about something else.